### PR TITLE
Prefix ad-related video events

### PR DESCRIPTION
**Pull request by dmlap**
Transform video events during ad playback to simplify the downstream event API. See https://github.com/videojs/video.js/issues/1825 for more details.

### CODE REVIEW

**Comment by bclwhitaker** [comment]
Please be sure to test that engagement is reported properly with the onceux and these changes (with prerolls, midrolls, and postrolls) -- There are some example streams for once in socrates.  I assume you are testing extensively with IMA already. There are a number of negative pre-roll examples in socrates that can offer quick validation that we're still handling errors proprerly.

File: example/example-integration.js [hunk 1]
```diff
@@ -14,22 +14,28 @@
...
       //  - adPlaying - whether a linear ad is currently playing
       //  - prerollPlayed - whether we've played a preroll
       //  - midrollPlayed - whether we've played a midroll
+      //  - postrollPlayed - whether we've played a postroll
       state = {},
-      
+
+      // just like any other video.js plugin, ad integrations can
+      // accept initialization options
+      adServerUrl = (options && options.adServerUrl) || "inventory.json",
```
Comment (seniorflexdeveloper) [inline example/example-integration.js]: Shouldn't this be (options && options.adServerUrl) ? options.adServerUrl : "inventory.json", or am I just reading wrong?

File: example/example-integration.js [hunk 2]
```diff
@@ -99,29 +107,29 @@
...
+
       if (state.midrollPlayed) {
         return;
       }
-      
+
       var currentTime = player.currentTime(), opportunity;
-      
+
       if ('lastTime' in state) {
```
Comment (seniorflexdeveloper) [inline example/example-integration.js]: Just curious, what is the 'lastTime' state for?

**Comment by seniorflexdeveloper** [comment]
@gkatsev - couple of semantics questions but functionally LGTM.

File: example/example-integration.js (hunk 1 above)
Comment (dmlap) [inline example/example-integration.js]: `&&` returns the second value when it succeeds, which is `options.adServerUrl` in this case. This line is being overly clever.

Comment (seniorflexdeveloper) [inline example/example-integration.js]: Got it, thx.

File: example/example-integration.js (hunk 2 above)
Comment (gkatsev) [inline example/example-integration.js]: I think this makes it play the midroll if you seek.

**Pull request closed without merging**